relay-spans/src/v2_to_v1.rs

```diff
@@ -270,6 +277,178 @@ fn derive_op_for_v2_span(span: &SpanV2) -> String {
+    // This is the ideal description.
+    if let Some(query_text) = attributes
+        .get("db.query.text")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(query_text.to_string());
+    }
+
+    // Other SDKs check for `db.statement`, it's a legacy OTel attribute, useful as a fallback in some cases.
+    if let Some(statement) = attributes
+        .get("db.statement")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(statement.to_string());
+    }
+
+    None
+}
+
+fn get_server_url_path(attributes: &Object<Attribute>) -> Option<String> {
+    // `http.route` takes precedence. If available, this is the matched route of the server
+    // framework for server spans. Not always available, even for server spans.
+    if let Some(route) = attributes
+        .get("http.route")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(route.to_string());
+    }
+
+    // `url.path` is the path of the HTTP request for server spans. This is required for server spans.
+    if let Some(path) = attributes
+        .get("url.path")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(path.to_string());
+    }
+
+    // `http.target` is deprecated, but might be present in older data. Here as a fallback
+    if let Some(target) = attributes
+        .get("http.target")
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        return Some(strip_url_query_and_fragment(target));
+    }
+
+    None
+}
+
+fn strip_url_query_and_fragment(url: &str) -> String {
+    url.split('?')
+        .next()
+        .unwrap_or(url)
+        .split('#')
+        .next()
+        .unwrap_or(url)
+        .to_string()
+}
+
+fn get_client_url_path(attributes: &Object<Attribute>) -> Option<String> {
+    // `url.full` is a new required attribute for client spans. This is the full URL and should be
+    // safe to parse. `http.url` is its old name
+    if let Some(url) = attributes
+        .get("url.full")
+        .or_else(|| attributes.get("http.url"))
+        .and_then(|attr| attr.value())
+        .and_then(|attr_val| attr_val.value.value.value())
+        .and_then(|v| v.as_str())
+    {
+        if let Ok(parsed_url) = Url::parse(url) {
+            return Some(format!(
+                "{}://{}{}",
+                parsed_url.scheme(),
+                parsed_url.domain().unwrap_or(""),
+                parsed_url.path()
+            ));
+        }
+    }
+
+    None
```

Comment:
In this function there is only the one attribute you're checking, so you don't need the `and_then` and `if let` rigmarole.
```suggestion
    let url = attributes
        .get("url.full")
        .or_else(|| attributes.get("http.url"))?
        .value()?
        .value
        .value
        .value()?
        .as_str()?;

    let parsed_url = Url::parse(url).ok()?;
    Some(format!(
        "{}://{}{}",
        parsed_url.scheme(),
        parsed_url.domain().unwrap_or(""),
        parsed_url.path()
    ))
```